examples/exchange-files-in-browser/public/app.js

```diff
@@ -165,33 +184,31 @@ function appendFile (name, hash, size, data) {
 
   $fileHistory.insertBefore(row, $fileHistory.firstChild)
 
-  publishHash(hash)
+  return publishHash(hash)
 }
 
-function getFile () {
-  const hash = $multihashInput.value
+async function getFile () {
+  const hash = $cidInput.value
 
-  $multihashInput.value = ''
+  $cidInput.value = ''
 
   if (!hash) {
-    return onError('No multihash was inserted.')
-  } else if (isFileInList(hash)) {
+    return onError('No CID was inserted.')
+  } else if (FILES.includes(hash)) {
     return onSuccess('The file is already in the current workspace.')
   }
 
   FILES.push(hash)
 
-  node.get(hash)
-    .then((files) => {
-      files.forEach((file) => {
-        if (file.content) {
-          appendFile(file.name, hash, file.size, file.content)
-          onSuccess(`The ${file.name} file was added.`)
-          $emptyRow.style.display = 'none'
-        }
-      })
-    })
-    .catch(() => onError('An error occurred when fetching the files.'))
+  const files = await node.get(hash)
+
+  return files.map(async (file) => {
```

Comment:
Yes, needs a `Promise.all` - my bad. It works (when there's no error) because code calling this function doesn't do anything with the result. Needs a fix.